common/src/KokkosKernels_helpers.hpp

```diff
@@ -95,6 +97,13 @@ struct GetUnifiedScalarViewType<T, TX, true, true> {
       type;
 };
 
+template <class... Ts>
+struct are_integral : std::bool_constant<((std::is_integral_v<Ts> ||
+                                           std::is_enum_v<Ts>)&&...)> {};
```

Comment:
This checks every template argument passed into are_integral_v and returns true if every type is either an integral or a enum?

Reply:
Yes.  See
https://github.com/kokkos/kokkos/blob/94d9c9e8b84d142f536c395d20925985c0f03f53/core/src/impl/Kokkos_Traits.hpp#L102-L106

Reply:
Why are you copying `Kokkos::Impl::are_integral` here?

Reply:
Showing you that the implementation I propose does not change the behavior.

Reply:
Or are you asking in general why moving functionality from Kokkos to KokkosKernels?  If so the answer is that it is in `Kokkos::Impl::` and that it is not legal to use, not even in KokkosKernels.

Reply:
Thanks, that answers my question.